network/endpoint_linux.go

```diff
@@ -234,34 +247,44 @@ func (nw *network) newEndpointImpl(_ apipaClient, nl netlink.NetlinkInterface, p
 		PODNameSpace:             epInfo.PODNameSpace,
 	}
 
+	if nw.extIf != nil {
+		ep.Gateways = []net.IP{nw.extIf.IPv4Gateway}
+	}
+
 	ep.Routes = append(ep.Routes, epInfo.Routes...)
 	return ep, nil
 }
 
 // deleteEndpointImpl deletes an existing endpoint from the network.
-func (nw *network) deleteEndpointImpl(nl netlink.NetlinkInterface, plc platform.ExecClient, ep *endpoint) error {
-	var epClient EndpointClient
-
+func (nw *network) deleteEndpointImpl(nl netlink.NetlinkInterface, plc platform.ExecClient, epClient EndpointClient, ep *endpoint) error {
 	// Delete the veth pair by deleting one of the peer interfaces.
 	// Deleting the host interface is more convenient since it does not require
 	// entering the container netns and hence works both for CNI and CNM.
-	if ep.VlanID != 0 {
-		epInfo := ep.getInfo()
-		if nw.Mode == opModeTransparentVlan {
-			log.Printf("Transparent vlan client")
-			epClient = NewTransparentVlanEndpointClient(nw, epInfo, ep.HostIfName, "", ep.VlanID, ep.LocalIP, nl, plc)
 
+	// epClient is nil only for unit test.
+	if epClient == nil {
+		//nolint:gocritic
+		if ep.VlanID != 0 {
+			epInfo := ep.getInfo()
+			if nw.Mode == opModeTransparentVlan {
+				log.Printf("Transparent vlan client")
+				epClient = NewTransparentVlanEndpointClient(nw, epInfo, ep.HostIfName, "", ep.VlanID, ep.LocalIP, nl, plc)
+
+			} else {
+				epClient = NewOVSEndpointClient(nw, epInfo, ep.HostIfName, "", ep.VlanID, ep.LocalIP, nl, ovsctl.NewOvsctl(), plc)
+			}
+		} else if nw.Mode != opModeTransparent {
+			epClient = NewLinuxBridgeEndpointClient(nw.extIf, ep.HostIfName, "", nw.Mode, nl, plc)
 		} else {
-			epClient = NewOVSEndpointClient(nw, epInfo, ep.HostIfName, "", ep.VlanID, ep.LocalIP, nl, ovsctl.NewOvsctl(), plc)
+			epClient = NewTransparentEndpointClient(nw.extIf, ep.HostIfName, "", nw.Mode, nl, plc)
 		}
-	} else if nw.Mode != opModeTransparent {
-		epClient = NewLinuxBridgeEndpointClient(nw.extIf, ep.HostIfName, "", nw.Mode, nl, plc)
-	} else {
-		epClient = NewTransparentEndpointClient(nw.extIf, ep.HostIfName, "", nw.Mode, nl, plc)
 	}
 
 	epClient.DeleteEndpointRules(ep)
-	epClient.DeleteEndpoints(ep)
+	// deleteHostVeth set to false not to delete veth as CRI will remove network namespace and
+	// veth will get removed as part of that.
+	//nolint:errcheck // ignore error
+	epClient.DeleteEndpoints(ep, false)
```

Comment:
If we pass this as `true` what will be the issue as it is going to be deleted in the future i suppose?

Reply:
right. not changing current behavior